### PR TITLE
Added connectathon test to avocado

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added connectathon Test case

### CODE REVIEW

**Comment by apahim** [comment]
Also here, job log please.

**Comment by PraveenPenguin** [comment]
@apahim Please find job log in below link 

https://paste.fedoraproject.org/401574/14703785/

Thanks

File: generic/connectathon.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                args = "-b -t"
+
+            process.system('./runtests -N %s %s %s' %
+                           (cthon_iterations, args, testdir), shell=True)
+
+        except:
+            self.nfail += 1
+            logging.error("Test failed: ")
+        if self.nfail != 0:
+            raise self.error('Connectathon test suite failed.')
```
Comment (apahim) [inline generic/connectathon.py:94]: `self.fail()`

File: generic/connectathon.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            self.error('Gcc is needed for the test to be run')
+        if not sm.check_installed("make") and not sm.install("make"):
+            self.error('make is needed for the test to be run')
+        if detected_distro.name == "SuSE":
+            if not sm.check_installed("git-core") and not sm.install("git-"
+                                                                     "core"):
+                self.error('git is needed for the test to be run')
+        else:
+            if not sm.check_installed("git") and not sm.install("git"):
+                self.error('git is needed for the test to be run')
```
Comment (apahim) [inline generic/connectathon.py:64]: Too much repetition on the same check. Could you simplify this? Something like this:
https://github.com/avocado-framework/avocado-misc-tests/blob/master/generic/gdb.py#L33-L43

**Comment by apahim** [comment]
Test works. Few minor improvements. Waiting v2. Thanks.

**Comment by PraveenPenguin** [comment]
@apahim  thanks , will adopt all the changes will open a new PR 

Regards
Praveen

**Comment by PraveenPenguin** [comment]
@apahim raise new PR #129 with fix , please have a look

Thanks

**Pull request closed without merging**